tools/nx-plugins/npm-package/executors/compile/executor.ts

```diff
@@ -0,0 +1,40 @@
+import path from 'node:path';
+import type { ExecutorContext } from '@nrwl/devkit';
+import { logger } from '@nrwl/devkit';
+import { rollupExecutor } from '@nrwl/rollup';
+import type { RollupExecutorOptions } from '@nrwl/rollup/src/executors/rollup/schema';
+import { setPkgDefaults } from '../utils/set-pkg-defaults';
+
+/**
+ * Compiles source code into an NPM package for publishing.
+ * Wraps @nrwl/rollup:rollup executor with some defaults set by us.
+ */
+export default async function (
+	options: RollupExecutorOptions,
+	context: ExecutorContext,
+) {
+	try {
+		// get some default from a rollup config
+		// it's a bit weird we have to do this, but this is how the original executor works
+		options.rollupConfig = path.resolve(__dirname, 'rollup.config.js');
```

Comment:
yeah this is a slightly confusing aspect of Nx writing plugins in TS es modules, but running them in cjs node. 

they're working on making nx a fully es module env, at which point this would need updating